### PR TITLE
Feature/pkg setup

### PR DESCRIPTION
**Pull request by joshwlambert**
Updates:

- Added tests to the package #10 & #13 
- Added continuous integration with GHA and code converage #11 & #12 
- Updated code style of package in line with tidyverse style guide #14 & #23 
- Couple of minor bug fixes from typos #16 & #17 

Please let me know if there is anything missing from the package in terms of basic setup.

### CODE REVIEW

File: R/epidist.R [hunk 1]
```diff
@@ -2,73 +2,121 @@
...
+
   # Extract study or default to largest sample size
-  if( is.null(study) ){pick_study <- pick_path %>% dplyr::filter(size==max(size))}
-  if(!is.null(study) ){pick_study <- pick_path %>% dplyr::filter(study_ID==study)}
+  if (is.null(study)) {
+    pick_study <- pick_path %>% dplyr::filter(size == max(size))
+  }
+  if (!is.null(study)) {
+    pick_study <- pick_path %>% dplyr::filter(study_ID == study)
+  }
```
Comment (thibautjombart) [inline R/epidist.R:57]: I would remove all instances of the piping operator. They make debugging really hard, I think the general recommendation is not to use them within packages.

File: R/extract_param.R [hunk 2]
```diff
@@ -1,151 +1,268 @@
 #' Extract parameters
 #'
 #' Extract parameters from summary data (e.g. in meta analysis)
-#' @param type A \code{character} defining whether summary statistics based around \code{percentiles} (default) or \code{range}
-#' @param values A \code{vector}. If \code{type=percentiles}: c(percentile_1, percentile_2); and if \code{type=range}: c(median, min, max)
-#' @param distribution A \code{character} specifying distribution to use. Default is \code{lnorm}; also takes \code{gamma} and \code{weibull}.
-#' @param percentiles A \code{vector} with two elements specifying the percentiles defined in \code{values} if using \code{type="percentiles"}.
-#' @param samples A \code{numeric} specifying the sample size if using \code{type="range"}.
+#' @param type A \code{character} defining whether summary statistics based
```
Comment (thibautjombart) [inline R/extract_param.R:4]: Same comment re doc as for epidist, now updated in [this issue](https://github.com/epiverse-trace/epiparameter/issues/25).

File: R/extract_param.R [hunk 3]
```diff
@@ -1,151 +1,268 @@
...
+#' around \code{percentiles} (default) or \code{range}
+#' @param values A \code{vector}. If \code{type=percentiles}: c(percentile_1,
+#' percentile_2); and if \code{type=range}: c(median, min, max)
+#' @param distribution A \code{character} specifying distribution to use.
+#' Default is \code{lnorm}; also takes \code{gamma} and \code{weibull}.
+#' @param percentiles A \code{vector} with two elements specifying the
+#' percentiles defined in \code{values} if using \code{type="percentiles"}.
+#' @param samples A \code{numeric} specifying the sample size if using
+#' \code{type="range"}.
 #' @keywords extract
```
Comment (thibautjombart) [inline R/extract_param.R:14]: Add an @author tag

**Review by thibautjombart — APPROVED** [review]
Thanks for the great work! I loved that the code was really easy to read and follow. I have not checked the math bits. 

I will accept the PR but raised 2 separate issues which are not urgent and are probably best addressed later:
* on [fleshing out the doc](https://github.com/epiverse-trace/epiparameter/issues/25) of key functions more
* on [removing piping operators](https://github.com/epiverse-trace/epiparameter/issues/26)

File: R/epidist.R (hunk 1 above)
Comment (joshwlambert) [reply R/epidist.R:57]: I agree, I had planned to do it after setting up the package, but more than happy to include it in this PR. I have now replaced the use of the tidyverse pipe with the base pipe, and deleted the import of the magrittr pipe. If you think that removal of the the base pipe is also recommended I am happy to go with that instead. What do you think?

File: R/extract_param.R (hunk 3 above)
Comment (joshwlambert) [reply R/extract_param.R:14]: Good spot, I've added author tags to each function. As I only made minor edits and formatting changes to the code I have not added myself as an author, happy to change this if needed.

File: tests/testthat/test-list_distributions.R [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  expect_snapshot(
+    list_distributions(type = "serial_interval", parameters = TRUE)
+  )
+})
+
+test_that("list_distributions fails correctly", {
+  # currently these do not throw an error
+  #expect_error(list_distributions(type = "incubatio", parameters = FALSE))
+  #expect_error(list_distributions(type = "incubatio", parameters = TRUE))
+})
```
[the remark this answers is not in the document]
Comment (joshwlambert) [reply tests/testthat/test-list_distributions.R:139]: I've made a note of this in an existing issues (#22) that tackles this.

**Pull request merged**